cdap-distributions/src/hdinsight/pageblob-configure.sh

```diff
@@ -0,0 +1,142 @@
+        die "[ERROR] Failed to update ${__configtype}: ${updateResult}"
+    fi
+}
+
+# Fetches value of fs.azure.page.blob.dir, appends '/cdap' to it, updates it via Ambari API, updates the local client configurations, and restarts cluster services
+updateFsAzurePageBlobDirForCDAP() {
+    currentValue=$(bash ${AMBARICONFIGS_SH} -u ${USERID} -p ${PASSWD} get ${ACTIVEAMBARIHOST} ${CLUSTERNAME} core-site | grep 'fs.azure.page.blob.dir' | cut -d' ' -f3 | sed -e 's/"\(.*\)"[,]*/\1/')
+    if [ -n ${currentValue} ]; then
+      if echo ${currentValue} | grep -q '/cdap'; then
+        echo "fs.azure.page.blob.dir already contains /cdap"
+        return 0
+      else
+        newValue="${currentValue},/cdap"
+      fi
+    else
+      newValue="/cdap"
+    fi
+    echo "Updating fs.azure.page.blob.dir to ${newValue}"
+    setAmbariConfig 'core-site' 'fs.azure.page.blob.dir' ${newValue} || die "Could not update Ambari config"
+    restartCdapDependentClusterServices
+}
+
+# Stop an Ambari cluster service
+stopServiceViaRest() {
+    if [ -z "${1}" ]; then
+        echo "Need service name to stop service"
+        exit 136
+    fi
+    SERVICENAME=${1}
+    echo "Stopping ${SERVICENAME}"
+    curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "INSTALLED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME}
+}
+
+# Start an Ambari cluster service
+startServiceViaRest() {
+    if [ -z "${1}" ]; then
+        echo "Need service name to start service"
+        exit 136
+    fi
+    sleep 2
+    SERVICENAME=${1}
+    echo "Starting ${SERVICENAME}"
+    startResult=$(curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "STARTED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME})
+    if [[ ${startResult} == *"500 Server Error"* || ${startResult} == *"internal system exception occurred"* ]]; then
+        sleep 60
+        echo "Retry starting ${SERVICENAME}"
+        startResult=$(curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "STARTED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME})
+    fi
+    echo ${startResult}
+}
+
+# Restart Ambari cluster services
+restartCdapDependentClusterServices() {
+    stopServiceViaRest HBASE
+    stopServiceViaRest YARN
+    stopServiceViaRest MAPREDUCE2
+    stopServiceViaRest HDFS
+
+    startServiceViaRest HDFS
+    startServiceViaRest MAPREDUCE2
+    startServiceViaRest YARN
+    startServiceViaRest HBASE
+}
```

Comment:
this I can't do anything about... the starts are asynchronous, and Ambari will not actually start them until this script itself completes.  I can't find any supporting documentation for this behavior, but it is consistent... I actually had to remove the status checking.

This restart is due to a change in core-site, so the other services should not be needed.  I may just try a full cluster restart though, since I am seeing cases where hbase doesn't come back up.